shell_automaton/src/protocol_runner/current_head/protocol_runner_current_head_state.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use crypto::hash::ContextHash;
+use serde::{Deserialize, Serialize};
+use tezos_protocol_ipc_client::ProtocolRunnerError;
+
+use crate::protocol_runner::ProtocolRunnerToken;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum ProtocolRunnerCurrentHeadState {
+    Init {
+        genesis_commit_hash: Option<ContextHash>,
```

Comment:
Why is this needed here? just so that it can be passed around for the final step? (because all these are not intermediary states before the initialization is complete)?

Reply:
I think it's to carry the value for `ProtocolRunnerState::Ready` state.

Reply:
It is kind of ugly, because on each step it then needs to be extracted and re-passed, but I guess there is no point in trying to change that now.

Reply:
@tizoc Yes this is to keep `genesis_commit_hash` until `ProtocolRunnerState::Ready` is reached.
Not sure what could be an alternative way to keep `genesis_commit_hash`